src/openssl.c

```diff
@@ -1191,6 +1192,31 @@ typedef void * (*pem_read_bio_func)(BIO *, void **, pem_password_cb *,
                                     void *u);
 #endif
 
+static int
+read_private_key_from_bio(void **key_ctx,
+                          pem_read_bio_func read_private_key,
+                          BIO *bp, const unsigned char *passphrase)
+{
+#ifdef HAVE_SSLERROR_BAD_DECRYPT
+    unsigned long sslError;
+#endif
+   *key_ctx = read_private_key(bp, NULL, (pem_password_cb *) passphrase_cb,
+                                (void *) LIBSSH2_UNCONST(passphrase));
+#ifdef HAVE_SSLERROR_BAD_DECRYPT
+   /* ensure to read pending openssl error (#1652) */
+   sslError = ERR_get_error();
+   if((ERR_GET_LIB(sslError) == ERR_LIB_PEM &&
+       ERR_GET_REASON(sslError) == PEM_R_BAD_DECRYPT) ||
+      (ERR_GET_LIB(sslError) == ERR_LIB_PROV &&
+       ERR_GET_REASON(sslError) == EVP_R_BAD_DECRYPT))
+       /* only error for incorrect passphrase to let caller fallback to openssh
+          format */
+      return _libssh2_error(NULL, LIBSSH2_ERROR_KEYFILE_AUTH_FAILED,
+                             "Wrong passphrase for private key");
```

Comment:
```suggestion
#ifdef HAVE_SSLERROR_BAD_DECRYPT
    /* ensure to read pending openssl error (#1652) */
    sslError = ERR_get_error();
    if((ERR_GET_LIB(sslError) == ERR_LIB_PEM &&
        ERR_GET_REASON(sslError) == PEM_R_BAD_DECRYPT) ||
       (ERR_GET_LIB(sslError) == ERR_LIB_PROV &&
        ERR_GET_REASON(sslError) == EVP_R_BAD_DECRYPT))
        /* only error for incorrect passphrase to let caller fallback to openssh
           format */
        return _libssh2_error(NULL, LIBSSH2_ERROR_KEYFILE_AUTH_FAILED,
                              "Wrong passphrase for private key");
```
fix indent, here too